### PR TITLE
cplp -1040: handle user creation case for already existing user.

### PR DESCRIPTION
**Pull request by Akshay-Chaudhari-BMW**
handle user creation case for already existing user

### CODE REVIEW

File: src/registration/CatenaX.NetworkServices.Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs [hunk 1]
```diff
@@ -154,6 +154,12 @@ public async Task SetCompanyWithAddressAsync(Guid applicationId, CompanyWithAddr
 
         public async Task<int> InviteNewUserAsync(Guid applicationId, UserCreationInfo userCreationInfo, string createdById)
         {
+            var userCheck = await _portalDBAccess.IsUserAlreadyExist(createdById);
+            if (userCheck)
+            {
+                throw new ForbiddenException($"user {createdById} is already exist");
```
Comment (jjeroch) [inline src/registration/CatenaX.NetworkServices.Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs:160]: @Akshay-Chaudhari-BMW very simple change - can we please update the error message to "user {createdById} does already exist"

Comment (Akshay-Chaudhari-BMW) [reply src/registration/CatenaX.NetworkServices.Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs]: resolved

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs [hunk 2]
```diff
@@ -578,6 +578,11 @@ public async IAsyncEnumerable<Guid> GetUserRoleIdsUntrackedAsync(IDictionary<str
                 .Include(companyApplication => companyApplication.Company)
                 .SingleOrDefaultAsync();
 
+        public async Task<bool> IsUserAlreadyExist(string iamUserId) =>
```
Comment (evegufy) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs:581]: I think 'IsUserExisting' as method name is better English.

File: src/registration/CatenaX.NetworkServices.Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs [hunk 3]
```diff
@@ -154,6 +154,12 @@ public async Task SetCompanyWithAddressAsync(Guid applicationId, CompanyWithAddr
 
         public async Task<int> InviteNewUserAsync(Guid applicationId, UserCreationInfo userCreationInfo, string createdById)
         {
+            var userCheck = await _portalDBAccess.IsUserAlreadyExist(createdById);
```
Comment (evegufy) [inline src/registration/CatenaX.NetworkServices.Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs:157]: I think 'userExists' as variable name is more straight forward.

**Review by evegufy — CHANGES_REQUESTED** [review]

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/IPortalBackendDBAccess.cs [hunk 4]
```diff
@@ -53,6 +53,7 @@ public interface IPortalBackendDBAccess
         Task<Invitation?> GetInvitationStatusAsync(string iamUserId);
         Task<RegistrationData?> GetRegistrationDataUntrackedAsync(Guid applicationId, string iamUserId);
         Task<CompanyApplication?> GetCompanyAndApplicationForSubmittedApplication(Guid applicationId);
+        Task<bool> IsUserAlreadyExist(string iamUserId);
```
[the remark this answers is not in the document]
Comment (Akshay-Chaudhari-BMW) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/IPortalBackendDBAccess.cs]: resolved

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs (hunk 2 above)
Comment (Akshay-Chaudhari-BMW) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/PortalBackendDBAccess.cs]: rename to IsUserExisting

File: src/registration/CatenaX.NetworkServices.Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs (hunk 3 above)
Comment (Akshay-Chaudhari-BMW) [reply src/registration/CatenaX.NetworkServices.Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs]: resolved

**Comment by evegufy** [comment]
@Akshay-Chaudhari-BMW Please resolve the conflicts, afterwards the merge is fine for me.
In general, please re-request the review once you made the changes, that makes the process easier with Github.
@jjeroch could you please also resolve your conversation?

**Review by evegufy — APPROVED** [review]

**Pull request merged**